src/navigators/RootNavigator.tsx

```diff
@@ -11,25 +11,145 @@ import {
+  FilterTransactionsScreen,
+  CreateAuthenticatorScreen,
+  CreateAuthenticatorPublicKeyScreen,
+  CreateAuthenticatorSuccessScreen,
+  DeleteEntityScreen,
+  ImportAuthenticatorScreen,
+  CreateWalletSuccessScreen,
+  IntegrateKeyScreen,
+  RecoveryTransactionListScreen,
+  RecoverySendScreen,
+  RecoverySeedScreen,
+  ImportWalletChooseTypeScreen,
+  OptionsAuthenticatorScreen,
+  ChunkedQrCode,
+  NotificationScreen,
+  AddEmailScreen,
+  ConfirmEmailScreen,
+  ChooseWalletsForNotificationScreen,
+  ChangeEmailScreen,
+  ReceiveNotificationsConfirmationScreen,
+  CreateTransactionPassword,
+  ConfirmTransactionPassword,
+  ConfirmNotificationCodeScreen,
+  AddNotificationEmailScreen,
 } from 'app/screens';
 
-import { MainCardStackNavigator } from './MainCardStackNavigator';
+import { MainTabNavigator } from './MainTabNavigator';
 
 const Stack = createStackNavigator<RootStackParams>();
 
-export const RootNavigator = () => (
-  <Stack.Navigator initialRouteName={Route.MainCardStackNavigator} headerMode="none">
-    <Stack.Screen name={Route.MainCardStackNavigator} component={MainCardStackNavigator} />
-    <Stack.Screen name={Route.ActionSheet} component={ActionSheet} options={modalOptions} />
-    <Stack.Screen name={Route.UnlockTransaction} component={UnlockTransaction} />
-    <Stack.Screen name={Route.EditText} component={EditTextScreen} />
-    <Stack.Screen name={Route.Message} component={MessageScreen} options={{}} />
-    <Stack.Screen name={Route.ExportWallet} component={ExportWalletScreen} />
-    <Stack.Screen name={Route.ExportWalletXpub} component={ExportWalletXpubScreen} />
-    <Stack.Screen name={Route.DeleteContact} component={DeleteContactScreen} />
-    <Stack.Screen name={Route.SendTransactionDetails} component={SendTransactionDetailsScreen} />
-  </Stack.Navigator>
-);
+interface Props {
+  shouldRenderOnBoarding: boolean;
+  shouldRenderNotification: boolean;
+}
+
+export const RootNavigator: FC<Props> = ({ shouldRenderOnBoarding, shouldRenderNotification }) => {
+  const getInitialRouteName = () => {
+    if (shouldRenderOnBoarding) {
+      return Route.CreatePin;
+    }
+    if (shouldRenderNotification) {
+      return Route.AddNotificationEmail;
+    }
+
+    return Route.MainTabStackNavigator;
+  };
+
+  return (
+    <Stack.Navigator initialRouteName={getInitialRouteName()} headerMode="none">
+      <Stack.Screen name={Route.CreatePin} component={CreatePinScreen} />
+      <Stack.Screen name={Route.ConfirmPin} component={ConfirmPinScreen} />
+      <Stack.Screen name={Route.CreateTransactionPassword} component={CreateTransactionPassword} />
+      <Stack.Screen name={Route.ConfirmTransactionPassword} component={ConfirmTransactionPassword} />
+
+      <Stack.Screen
+        name={Route.AddNotificationEmail}
+        component={AddNotificationEmailScreen}
+        options={{ gestureEnabled: false }}
+      />
+      <Stack.Screen name={Route.ConfirmNotificationCode} component={ConfirmNotificationCodeScreen} />
```

Comment:
ConfirmNotificationCode  is duplicated with ConfirmEmailScreen, same as AddNotificationEmail and AddEmailScreen
It would be better to reuse them